src/java/org/jivesoftware/openfire/plugin/rest/AuthFilter.java

```diff
@@ -33,15 +39,13 @@
     private RESTServicePlugin plugin = (RESTServicePlugin) XMPPServer.getInstance().getPluginManager()
             .getPlugin("restapi");
 
-    /*
-     * (non-Javadoc)
-     * 
-     * @see
-     * com.sun.jersey.spi.container.ContainerRequestFilter#filter(com.sun.jersey
-     * .spi.container.ContainerRequest)
-     */
     @Override
-    public ContainerRequest filter(ContainerRequest containerRequest) throws WebApplicationException {
+    public void filter(ContainerRequestContext containerRequest) throws IOException {
+        if (containerRequest.getUriInfo().getRequestUri().getPath().equals("/plugins/restapi/v1/openapi.yaml")) {
```

Comment:
Should we exclude it? In swagger UI we can set the secret key. I think we should make fewer "exceptions" in the authentication part, which is crucial.